src/completions/index.ts

```diff
@@ -0,0 +1,303 @@
+import * as fs from 'fs-extra';
+import * as path from 'path';
+
+import { checkInstalledVersion } from '../updates/appc/core';
+import { CustomError } from '../utils';
+
+import { homedir } from 'os';
+/**
+ * Load completions list
+ *
+ * @returns {Object}
+ */
+export async function loadCompletions (sdk: string) {
+	try {
+		const alloyVersion = await getAlloyVersion();
+		const sdkCompletions = await fs.readJSON(getSDKCompletionsFileName(sdk));
+		const alloyCompletions = await fs.readJSON(getAlloyCompletionsFileName(alloyVersion));
+		Object.assign(sdkCompletions.properties, {
+			id: {
+				description: 'TSS id'
+			},
+			class: {
+				description: 'TSS class'
+			},
+			platform: {
+				type: 'String',
+				description: 'Platform condition',
+				values: [
+					'android',
+					'ios',
+					'mobileweb',
+					'windows'
+				]
+			}
+		});
+		return {
+			alloy: alloyCompletions,
+			titanium: sdkCompletions
+		};
+	} catch (error) {
+		throw error;
+	}
+}
+
+export function getSDKCompletionsFileName (version: string) {
+	return path.join(homedir(), '.titanium', 'completions', 'titanium', version, 'completions.json');
+}
+
+export function getAlloyCompletionsFileName (version: string) {
+	return path.join(homedir(), '.titanium', 'completions', 'alloy', version, 'completions.json');
+}
+
+/**
+ * Generate completions for an Alloy version.
+ *
+ * @param {Boolean} force - Force generation of completion file.
+ */
+export async function generateAlloyCompletions (force = false) {
+	const appcPath = path.join(homedir(), '.appcelerator', 'install');
+	const version = await checkInstalledVersion();
+	if (!version) {
+		throw Error('Unable to find installed alloy version.');
+	}
+	const alloyPath = path.join(appcPath, version, 'package', 'node_modules', 'alloy');
+	const alloyVersion = await getAlloyVersion();
+
+	const alloyCompletionsFilename = getAlloyCompletionsFileName(alloyVersion);
+
+	if (!force && await fs.pathExists(alloyCompletionsFilename)) {
```

Comment:
Just had a thought, we include a version in the files. We probably want to add that into the checks here and on line 154 so that's usable going forward